packages/vscode/src/hashes.ts

```diff
@@ -0,0 +1,36 @@
+import crypto from 'crypto'
+import { window, workspace } from 'vscode'
+
+/**
+ * Get a unique identifier for the project by hashing
+ * the directory with `schema.prisma`
+ */
+export async function getProjectHash(): Promise<string> {
+
+    let projectPath = await getSchemaPath()
+    projectPath = projectPath || process.cwd() // Default to cwd if the schema couldn't be found
+    console.log('projectPath: ' + projectPath)
+
+    return crypto
+        .createHash('sha256')
+        .update(projectPath)
+        .digest('hex')
+        .substring(0, 8)
+}
+
+
+async function getSchemaPath(): Promise<string | null> {
+    // try the currently open document
+    let schemaPath = window.activeTextEditor?.document.fileName
+    if (schemaPath && schemaPath.endsWith('.prisma')) {
+      return schemaPath
+    }
```

Comment:
When can it happen that this is called but the current open document is not a `schema.prisma` file?

Reply:
If you're not in a Prisma project for example and just trying out the schema with a separate file for example